add relation examples for {{this}} (#4011)
fishtown-analytics/docs.getdbt.com: website/docs/reference/dbt-classes.md

The `Relation` object is used to interpolate schema and <Term id="table" /> names into SQL code with appropriate quoting. This object should _always_ be used instead of interpolating values with `{{ schema }}.{{ table }}` directly. Quoting of the Relation object can be configured using the [`quoting` config](/reference/project-configs/quoting).

> dbt provides built-in relations, such as [`this`](/reference/dbt-jinja-functions/this), [`source`](/reference/dbt-jinja-functions/source), [`ref`](/reference/dbt-jinja-functions/ref).

Reviewer: ```suggestion
```
Rephrased this and moved this to the next section

### Using relations

> Use relations in your SQL code by wrapping them in double curly brackets `{{ }}` (such as the [Jinja function {{ this }}](/reference/dbt-jinja-functions/this)). For example:
> 

Reviewer: ```suggestion
In addition to `api.Relation.create`, dbt returns a Relation when you use [`ref`](/reference/dbt-jinja-functions/ref), [`source`](/reference/dbt-jinja-functions/source) or  [`this`](/reference/dbt-jinja-functions/this).
```

This isn't wrong, but it's not precise so I think it'll build an inaccurate mental model for people. It also conflicts with the code sample below, where the relation is created inside of a `{%  %} block not `{{ }}`.
